test/uk/gov/hmrc/cdsimportsddsfrontend/services/DeclarationXml_DocumentationSpec.scala

```diff
@@ -51,4 +51,51 @@ class DeclarationXml_DocumentationSpec extends WordSpec with MustMatchers {
     }
   }
 
+  "Previous documents data" should {
+    "not be populated in the XML when previous document is empty" in {
+
+      val documentationEmpty = DocumentationType().copy(previousDocument = Seq.empty)
+      val declaration = Declaration().copy(documentationType = documentationEmpty)
+
+      val xmlElement: Elem = (new DeclarationXml).fromImportDeclaration(declaration)
+      val governmentAgencyGoodsItem = (xmlElement \ "Declaration" \ "GoodsShipment" \ "GovernmentAgencyGoodsItem")
+
+      (governmentAgencyGoodsItem \ "PreviousDocument").toList.size mustBe 0
+    }
+  }
+
+  "Previous documents CategoryCode data" should {
+    "not be created in the XML when it is None" in {
+
+      val previousDocumentsWithNoCategory = PreviousDocument().copy(categoryCode = None)
+
+      val documentation = DocumentationType().copy(previousDocument = Seq(previousDocumentsWithNoCategory))
+      val declaration = Declaration().copy(documentationType = documentation)
+
+      val xmlElement: Elem = (new DeclarationXml).fromImportDeclaration(declaration)
+      val governmentAgencyGoodsItem = (xmlElement \ "Declaration" \ "GoodsShipment" \ "GovernmentAgencyGoodsItem")
+
+      (governmentAgencyGoodsItem \ "PreviousDocument" \ "CategoryCode").toList.size mustBe 0
+      (governmentAgencyGoodsItem \ "PreviousDocument" \ "TypeCode").toList.map(_.text) must contain("CLE")
```

Comment:
It would be clearer to test the whole list of expected items here, "CLE", "DCR", "ZZZ" etc. Same comment for the checks below